ddtrace/tracer/tracer.go

```diff
@@ -453,15 +453,9 @@ func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOpt
 		}
 	}
 	span.context = newSpanContext(span, context)
-	if context == nil || context.span == nil {
-		// this is either a root span or it has a remote parent, we should add the PID.
-		span.setMeta(ext.Pid, t.pid)
-		if _, ok := opts.Tags[ext.ServiceName]; !ok && t.config.runtimeMetrics {
-			// this is a root span in the global service; runtime metrics should
-			// be linked to it:
-			span.setMeta("language", "go")
-		}
-	}
+	span.setMetric(ext.Pid, float64(t.pid))
+	span.setMeta("language", "go")
+
```

Comment:
We could use some information about why this is changing, as it doesn't seem to be related to the broader opentelemetry-related tag stuff.

* Why are we now setting pid/language tags on all spans?
* Why is the pid now a metric rather than metadata?
* Why is pid changing from `system.pid` to `process_id`?

Reply:
process_id needs to be a numeric value, and since span.setMeta only sets string values, we have to use span.setMetric to set the process_id as a numeric value

Reply:
I'm definitely lacking context here, just want to make sure these two questions aren't about breaking changes. Also would this require a change in the trace-agent? Should we keep both fields instead of replacing?

> Why is the pid now a metric rather than metadata?
Why is pid changing from system.pid to process_id?

Would you confirm @knusbaum ?

Reply:
Another detail about the system.pid, even though we are just renaming it with this PR, the backend has already been remapping and duplicating the value "system.pid" to "process.id" so it wouldn't produce any breaking changes. Renaming the tag and changing its value is fine as other tracers already have numerical process_id tags.

Reply:
This doesn't answer the numeric change. Are we sure this needs to be a metric and not metadata? It was previously metadata. Fields like this that are used to connect things sometimes are very particular. 

I also still have concerns (put in slack) about this change, as it doesn't seem to match the spec, and I'm hoping we won't need to do this at all.